orangecontrib/text/widgets/owpubmed.py

```diff
@@ -0,0 +1,463 @@
+import os
+import re
+
+from PyQt4 import QtGui, QtCore
+from PyQt4.QtGui import *
+from validate_email import validate_email
+
+from Orange.widgets import gui
+from Orange.widgets.settings import Setting
+from Orange.widgets.widget import OWWidget
+from orangecontrib.text.corpus import Corpus
+from orangecontrib.text.pubmed import (
+    Pubmed, PUBMED_TEXT_FIELDS
+)
+
+
+def _i(name, icon_path='icons'):
+    widget_path = os.path.dirname(os.path.abspath(__file__))
+    return os.path.join(widget_path, icon_path, name)
+
+
+def eval_date(date):
+    if not date:
+        return True
+    pattern = re.compile('\d{4}/\d{2}/\d{2}')
+    return pattern.match(date)
+
+
+class Output:
+    CORPUS = 'Corpus'
+
+
+class OWPubmed(OWWidget):
+    name = 'Pubmed'
+    description = 'Load data from the Pubmed api.'
+    icon = 'icons/Pubmed.svg'
+    priority = 20
+
+    outputs = [(Output.CORPUS, Corpus)]
+    want_main_area = False
+
+    # Settings.
+    recent_emails = Setting([])
+    author = Setting('')
+    pub_date_from = Setting('')
+    pub_date_to = Setting('')
+    recent_keywords = Setting([])
+    last_advanced_query = Setting('')
+    num_records = Setting(1000)
+
+    # Text includes checkboxes.
+    includes_authors = Setting(False)
+    includes_title = Setting(False)
+    includes_mesh = Setting(False)
+    includes_abstract = Setting(True)
+
+    def __init__(self):
+        super().__init__()
+
+        self.output_corpus = None
+        self.pubmed_api = None
+        self.progress = None
+        self.email_is_valid = False
+        self.record_count = 0
+
+        # To hold all the controls. Makes access easier.
+        self.pubmed_controls = []
+
+        h_box = gui.widgetBox(self.controlArea, orientation='horizontal')
```

Comment:
`orientation='horizontal'` and `orientation='vertical'` are depricated as of a few commits ago on orange master. Please, use `Qt.Horizontal` and `Qt.Vertical` instead. In this particular example it would be even better to just opt for `gui.hBox`.